proxy/upgrade_test.go

```diff
@@ -412,18 +412,18 @@ func TestAuditLogging(t *testing.T) {
 				<-p.auditLogHook
 			}
 
-			check(t, sout, serr)
+			check(t, sout.String(), serr)
 		}
 	}
 
-	t.Run("off", test(false, func(t *testing.T, sout, serr *bytes.Buffer) {
-		if sout.Len() != 0 || serr.Len() != 0 {
+	t.Run("off", test(false, func(t *testing.T, sout string, serr *bytes.Buffer) {
+		if sout != "<nil>" || serr.Len() != 0 {
```

Comment:
I think empty buffer will produce empty string, not `<nil>`

Reply:
Somehow I saw in the stdlib code `<nil>` but the test shows `""`, likely because bytes.Buffer was initialized and not nil itself.